Source/SwiftLintCore/Extensions/Configuration+Parsing.swift

```diff
@@ -147,35 +152,115 @@ extension Configuration {
     }
 
     private static func validateConfiguredRulesAreEnabled(
+        parentConfiguration: Configuration?,
         configurationDictionary dict: [String: Any],
         ruleList: RuleList,
         rulesMode: RulesMode
     ) {
         for key in dict.keys where !validGlobalKeys.contains(key) {
             guard let identifier = ruleList.identifier(for: key),
-                let rule = ruleList.list[identifier] else {
+                let ruleType = ruleList.list[identifier] else {
                     continue
             }
 
-            let message = "Found a configuration for '\(identifier)' rule"
-
             switch rulesMode {
             case .allEnabled:
                 return
-
             case .only(let onlyRules):
-                if Set(onlyRules).isDisjoint(with: rule.description.allIdentifiers) {
-                    Issue.genericWarning("\(message), but it is not present on '\(Key.onlyRules.rawValue)'.").print()
-                }
-
+                let issue = validateConfiguredRuleIsEnabled(onlyRules: onlyRules, ruleType: ruleType)
+                issue?.print()
             case let .default(disabled: disabledRules, optIn: optInRules):
-                if rule is any OptInRule.Type, Set(optInRules).isDisjoint(with: rule.description.allIdentifiers) {
-                    Issue.genericWarning("\(message), but it is not enabled on '\(Key.optInRules.rawValue)'.").print()
-                } else if Set(disabledRules).isSuperset(of: rule.description.allIdentifiers) {
-                    Issue.genericWarning("\(message), but it is disabled on '\(Key.disabledRules.rawValue)'.").print()
+                let issue = validateConfiguredRuleIsEnabled(
+                    parentConfiguration: parentConfiguration,
+                    disabledRules: disabledRules,
+                    optInRules: optInRules,
+                    ruleType: ruleType
+                )
+                issue?.print()
+            }
+        }
+    }
+
+    static func validateConfiguredRuleIsEnabled(
+        parentConfiguration: Configuration?,
+        disabledRules: Set<String>,
+        optInRules: Set<String>,
+        ruleType: any Rule.Type
+    ) -> Issue? {
+        var enabledInParentRules: Set<String> = []
+        var disabledInParentRules: Set<String> = []
+        var allEnabledRules: Set<String> = []
+
+        if case .only(let onlyRules) = parentConfiguration?.rulesMode {
+            enabledInParentRules = onlyRules
+        } else if case .default(let parentDisabledRules, let parentOptInRules) = parentConfiguration?.rulesMode {
+            enabledInParentRules = parentOptInRules
+            disabledInParentRules = parentDisabledRules
+        }
+        allEnabledRules = enabledInParentRules
+            .subtracting(disabledInParentRules)
+            .union(optInRules)
+            .subtracting(disabledRules)
```

Comment:
What about rules enabled by default? Are they included somewhere else?

Reply:
That is a good point. I *think* they're covered, but there may be some cases I haven't thought of.

Reply:
Yep, they should work ok, and there are test cases for them.

If a rule is not present in `allEnabledRules`, then we will look to see if it is disabled in the parent or child. If not, and it's a default rule, then we only report an issue if the configuration uses only rules, and our rule is not present there.